pallets/gear/src/manager/mod.rs

```diff
@@ -367,27 +366,14 @@ where
         }
     }
 
-    /// Removes memory pages of the program and transfers program balance to the `value_destination`.
+    /// Removes reservation map and memory pages of the program
     fn clean_inactive_program(
         program_id: ProgramId,
-        memory_infix: MemoryInfix,
-        value_destination: ProgramId,
+        program: &mut ActiveProgram<BlockNumberFor<T>>,
     ) {
-        ProgramStorageOf::<T>::remove_program_pages(program_id, memory_infix);
-
-        let program_account = program_id.cast();
-        let balance = CurrencyOf::<T>::free_balance(&program_account);
-        if !balance.is_zero() {
-            let destination = Pallet::<T>::inheritor_for(value_destination).cast();
-
-            CurrencyOf::<T>::transfer(
```

Comment:
revert this part.
on exit we transfer balance to destination, but only once
any other balance remaining will be transferred through new extrinsic